digdag-standards/src/main/java/io/digdag/standards/command/EcsCommandExecutor.java

```diff
@@ -321,13 +322,16 @@ CommandStatus createNextCommandStatus(
 
         // To fetch log until all logs is written in CloudWatch, it should wait until getting finish marker in end of task.
         // (finished previous poll once considering risk of crushing while running previous actual poll.)
-        final Optional<Long> taskFinishedAt = !previousStatus.has("task_finished_at") ?
-                Optional.absent() : Optional.of(previousStatus.get("task_finished_at").asLong());
-        if (taskFinishedAt.isPresent()) {
-            long timeout = taskFinishedAt.get() + 60;
+        if (previousStatus.has("task_finished_at")) {
+            final long taskFinishedAt = previousStatus.get("task_finished_at").asLong();
+            final int statusCode =  previousStatus.get("status_code").intValue();
+
+            boolean foundLoggingFinishedMark = false;
```

Comment:
I don't get why missing ECS_END_OF_TASK_LOG_MARK can be the cause of this issue. An ECS task should be finished in this `if` block since the status has `task_finished_at` meaning all the commands built in `setEcsContainerOverrideCommand()` are done. Missing ECS_END_OF_TASK_LOG_MARK might happen due to (possible?) visibility delay of CloudWatch log, but the `tar -zcf` command should be done no matter whether it finished successfully.

Reply:
ECS tasks can fail before https://github.com/treasure-data/digdag/blob/v0_11/digdag-standards/src/main/java/io/digdag/standards/command/EcsCommandExecutor.java#L750
and it's accepted by caller.
https://github.com/treasure-data/digdag/blob/42ea7cb71270957ea6424c2df88dd7c65d248789/digdag-standards/src/main/java/io/digdag/standards/operator/PyOperatorFactory.java#L169

Reply:
Some tasks does not have `archive-output.tar.gz` in s3.  That's the issue. This can be happens when docker process abnormally exits by some reasons. ECS_END_OF_TASK_LOG_MARK indicate archive-output.tar.gz exists in s3.

Also, note python process abnormally exits without dumping output json.
https://github.com/treasure-data/digdag/blob/master/digdag-standards/src/main/resources/digdag/standards/py/runner.py#L190

Reply:
please also not this modification does not affect normal cases where `archive-output.tar.gz` exists.

Reply:
> ECS tasks can fail before https://github.com/treasure-data/digdag/blob/v0_11/digdag-standards/src/main/java/io/digdag/standards/command/EcsCommandExecutor.java#L750
and it's accepted by caller.

Does it mean retrying after catching StorageFileNotFoundException is meaningless in this case?

Reply:
yes, need to throw RuntimeException then as same as the current behavior.

https://github.com/treasure-data/digdag/pull/1603/files#diff-91bd8388edfb595c57a91f7eb0966d7b2548965a3a7e0fc3f374780255bfab3aR377

Reply:
however, exit non-zero handling would be useful because it's suspected some cases for no archive-output.tar.gz are related to exit non-zero.
https://github.com/treasure-data/digdag/pull/1603/files#diff-91bd8388edfb595c57a91f7eb0966d7b2548965a3a7e0fc3f374780255bfab3aR382

Reply:
I thought `ECS tasks can fail before` meant the following situation since I knew this PR considering `exit $exit_code`.
- The custom script itself finished successfully
- But other command (e.g. `tar zcf` or `curl -X PUT`) failed

My question is what if the above case happens. The status code is 0, but `archive-output.tar.gz` doesn't appear forever.

Reply:
```
                    if (foundLoggingFinishedMark == false && previousStatus.has("retry_on_end_of_task_mark_missing") == false) {
                        logger.warn(s("Scheduled to poll again because "
                                + "1) archive-output.tar.gz does not exist yet while container task normally exit 0, and "
                                + "2) no ECS_END_OF_TASK_LOG_MARK observed yet. "
                                + "cluster=%s, taskArn=%s, errorMessage=%s", cluster, taskArn, errorMessage.orNull()), ex);
                        nextStatus.put("retry_on_end_of_task_mark_missing", true);
                        return EcsCommandStatus.of(false, nextStatus); // poll again
                    } else {
```

can be removed while it's for problem investigation for unforeseen issues.

Reply:
Followed your comments https://github.com/treasure-data/digdag/pull/1603/commits/e11eacccc60d89847bfc01ee3896173d6dd894dd